### PR TITLE
Add zsh completion for 'docker network'

### PR DESCRIPTION
**Pull request by sdurrheimer**
Zsh completion for docker network and subcommands.

@vincentbernat 

Signed-off-by: Steve Durrheimer s.durrheimer@gmail.com

### CODE REVIEW

File: contrib/completion/zsh/_docker [hunk 1]
```diff
@@ -335,7 +428,7 @@ __docker_subcommand() {
         "($help)*--lxc-conf=-[Add custom lxc options]:lxc options: "
         "($help)--mac-address=-[Container MAC address]:MAC address: "
         "($help)--name=-[Container name]:name: "
-        "($help)--net=-[Network mode]:network mode:(bridge none container host)"
+        "($help)--net=-[Connects a container to a network]:network mode:(bridge none container host)"
```
Comment (vincentbernat) [inline contrib/completion/zsh/_docker]: Nitpick: no "s" for "connect"

File: contrib/completion/zsh/_docker [hunk 2]
```diff
@@ -605,6 +698,23 @@ __docker_subcommand() {
...
+                $opts_help \
+                "($help -): :->command" \
+                "($help -)*:: :->option-or-argument" && ret=0
+
+            case $state in
+                (command)
+                    __docker_network_commands && ret=0
+                    ;;
+                (option-or-argument)
+                    curcontext=${curcontext%:*:*}:docker-$words[-1]:
```
Comment (vincentbernat) [inline contrib/completion/zsh/_docker]: I suppose it works as is, but I would have put braces: `${words[-1]}`.

**Comment by vincentbernat** [comment]
LGTM. Great work!

**Comment by jfrazelle** [comment]
thanks @vincentbernat for reviewing! LGTM

**Pull request merged**